optional_plugins/tap/avocado_tap/__init__.py

```diff
@@ -0,0 +1,315 @@
+
+        if result.exit_status != 0:
+            self.fail('TAP Test execution returned a '
+                      'non-0 exit code (%s)' % result)
+        parser = TapParser(io.StringIO(result.stdout_text))
+        fail = 0
+        count = 0
+        bad_errormsg = 'there were test failures'
+        for event in parser.parse():
+            if isinstance(event, TapParser.Error):
+                self.error('TAP parsing error: ' + event.message)
+                continue
+            if isinstance(event, TapParser.Bailout):
+                self.error(event.message)
+                continue
+            if isinstance(event, TapParser.Test):
+                bad = event.result in (TestResult.XPASS, TestResult.FAIL)
+                if event.result != TestResult.SKIP:
+                    count += 1
+                if bad:
+                    self.log.error('%s %s %s', event.result.name, event.number, event.name)
+                    fail += 1
+                    if event.result == TestResult.XPASS:
+                        bad_errormsg = 'there were test failures or unexpected passes'
+                else:
+                    self.log.info('%s %s %s', event.result.name, event.number, event.name)
+
+        if not count:
+            raise exceptions.TestSkipError('no tests were run')
+        if fail:
+            self.fail(bad_errormsg)
+
+
+class TapLoader(loader.SimpleFileLoader):
+    """
+    Test Anything Protocol loader class
+    """
+    name = "tap"
+
+    @staticmethod
+    def get_type_label_mapping():
+        mapping = loader.SimpleFileLoader.get_type_label_mapping()
+        mapping.update(
+            {TapTest: 'TAP'})
+        return mapping
+
+    @staticmethod
+    def get_decorator_mapping():
+        mapping = loader.SimpleFileLoader.get_decorator_mapping()
+        mapping.update(
+            {TapTest: output.TERM_SUPPORT.healthy_str})
+        return mapping
+
+    def _make_simple_test(self, test_path, subtests_filter):
+        return self._make_test(TapTest, test_path,
+                               subtests_filter=subtests_filter,
+                               executable=test_path)
+
+
+class TapResolver(Resolver):
+
+    name = 'tap'
+    description = 'Test resolver for executable files to be handled as tests'
+
+    @staticmethod
+    def resolve(reference):
+
+        criteria_check = check_file(reference, reference, suffix=None,
+                                    type_name='executable file',
+                                    access_check=os.R_OK | os.X_OK,
+                                    access_name='executable')
+        if criteria_check is not True:
+            return criteria_check
+
+        return ReferenceResolution(reference,
+                                   ReferenceResolutionResult.SUCCESS,
+                                   [Runnable('tap', reference)])
+
+
+class TapCLI(CLI):
+
+    """
+    Run Test Anything Protocol tests
+    """
+
+    name = 'tap'
```

Comment:
@clebergnu or @bonzini, we already have a `name = 'TAP'` from CLI on `plugins/tap.py`.

Will this name conflict with the current plugin or not?

Reply:
This name is purely informational, so it won't cause a hard conflict other than possibly confusing users about two plugins of the same type and same name.  The confusion is highly minimized by the different descriptions, though.

Reply:
Yes, I was talking about a "soft conflict", two plugins with the same name.

Reply:
Yes, no conflict is better than a soft conflict indeed.  If possible, we should rename those, such as `tap_results` and `tap_loader` IMO.